services/src/main/java/com/linkedin/kmf/services/DefaultMetricsReporterService.java

```diff
@@ -32,13 +38,19 @@
   private final List<String> _metricNames;
   private final int _reportIntervalSec;
   private final ScheduledExecutorService _executor;
+  private final GraphiteClient _graphiteClient;
+  private final String _metricNamePrefix;
 
-  public DefaultMetricsReporterService(Map<String, Object> props, String name) {
+  public DefaultMetricsReporterService(Map<String, Object> props, String name) throws SocketException, UnknownHostException {
```

Comment:
I think you are changing `DefaultMetricsReporterService` so that it output data to both log4j and graphite. It means that user will have to use graphte even if they only needs to see metrics in the log4j output. Can you write e.g. `GraphteMetricsReporterService` that implements the `Service` interface to export metrics to graphite? Maybe the graphte-related configs can be put in `GraphteMetricsReporterServiceConfig`.